File: sdks/remote_plugin/python/app/core/flow.py
from __future__ import annotations

from abc import ABC, abstractmethod

from app.schema import Payload, Signal


class Flow(ABC):
    """Interface used by plugins to emit payloads and signals."""

    @abstractmethod
    async def send_payload(
        self,
        payload: Payload,
        *,
        port: int = 0,
    ) -> None:
        pass

    @abstractmethod
    async def send_signal(
        self,
        signal: Signal,
        *,
        port: int = 0,
    ) -> None:
        pass

File: sdks/remote_plugin/nodejs/src/core/flow.ts
import type { Signal } from '../schema/signal.js'
import type { Payload } from '../schema/payload.js'

/**
 * Flow interface used by plugins to emit signals and payloads back into the DAG.
 */
export interface Flow {
  sendPayload(payload: Payload, options?: { port?: number }): Promise<void>
  sendSignal(signal: Signal, options?: { port?: number }): Promise<void>
}
